scripts/publish.js

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable camelcase */
+import inquirer from "inquirer";
+import { invalid, section, valid } from "../utils.js";
+import ora from "ora";
+import { apiClient } from "./utils/apiClient.js";
+
+const POLL_INTERVAL = 2000;
+
+const IN_PROGRESS_STATUSES = ["PENDING", "STARTED"];
+
+const waitForTaskToComplete = async (taskId) => {
+  const waitingSpinner = ora(
+    "Waiting for publish to complete. This may take a few minutes.\n\n Publishing will continue if you exit here (CTRL+C)."
+  ).start();
+
+  let latestProcessingResultBody;
+
+  // Poll indefinitely until the task is resolved.
+  while (true) {
+    await new Promise((resolve) => {
+      setTimeout(() => {
+        resolve();
+      }, POLL_INTERVAL);
+    });
+
+    const processingResult = await apiClient.get({
+      path: "/v2/status-of-published-module/" + taskId
+    });
+
+    if (!processingResult.ok) {
+      waitingSpinner.stop();
+
+      // TODO - ask the BE to supply a 400 when there's a regular failure?
+      if (processingResult.status === 500) {
```

Comment:
It would be nice if failed but expected tasks "such as malformed URL, or no permissions" returned with a 200 (or 400). That way we can distinguish between expected and unexpected errors on the CLI cleaner.